Frameworks/CoreText/CTRun.mm

```diff
@@ -79,101 +88,122 @@ CTRunStatus CTRunGetStatus(CTRunRef run) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 const CGGlyph* CTRunGetGlyphsPtr(CTRunRef run) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    _CTRun* runPtr = static_cast<_CTRun*>(run);
+    return runPtr && runPtr->_glyphs.size() ? runPtr->_glyphs.data() : NULL;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 void CTRunGetGlyphs(CTRunRef run, CFRange range, CGGlyph buffer[]) {
-    UNIMPLEMENTED();
+    _CTRun* runPtr = static_cast<_CTRun*>(run);
+    if (runPtr) {
+        if (range.location < 0) {
```

Comment:
You're actually repeating this pattern a lot in this file - make a helper for it?

---
In reply to: [78249297](https://github.com/Microsoft/WinObjC/pull/989#discussion_r78249297) [](ancestors = 78249297)